overlord/servicestate/quota_control.go

```diff
@@ -306,6 +306,14 @@ func CreateQuota(st *state.State, name string, parentName string, snaps []string
 		return err
 	}
 
+	// make sure the memory limit is at least 4K, that is the minimum size
+	// to allow nesting, otherwise groups with less than 4K will trigger the
+	// oom killer to be invoked when a new group is added as a sub-group to the
+	// larger group.
+	if memoryLimit <= 4*quantity.SizeKiB {
```

Comment:
Hm I guess a quota the size of a single page is not very useful either :rofl: but alas, if one wants to use and learn the hard way I don't see a reason to prevent that.

Reply:
I assume with this 4k you can add multiple sub-groups as well? It's not 4k for each sub-group? If so, then :+1:

Reply:
ah well actually funny thing you should ask ... no it's 4K _per_ group it seems but we can't know ahead of time how many sub-groups may go into a group. We could however add accounting to ensure that if a child group is being added that the parent has at least 4096 leftover

Reply:
For now the thing we will do is to just make the minimum 4K and if folks disagree or have issues we can increase it or make other adjustments later.